enemy.gd

```diff
@@ -0,0 +1,71 @@
+extends CharacterBody2D
+class_name BasicEnemy
+
+const speed: float = 50 #1/4 as fast as the player
+var is_chase: bool = true #Is the enemy currently chasing the player? Will only chase if theyre close
+var is_roaming: bool  = false #Wander around when not attacking the player
+var dead: bool = false
+var taking_damage: bool = false #Halt other animations/actions when attacked
+var is_dealing_damage: bool = false #Trigger damage animation
+var dir: Vector2 #He can go left or right no jumping
+const gravity = 900 
+var knockback_force = -20 #When hit, the enemy will be knocked backwards
+@export var player: CharacterBody2D #Remember to set player in scene inspector
+var player_in_range = false #Is the player close enough to chase
+@onready var damage_zone = $DealDamageArea
+const chase_distance: float = 100 #Distance where enemy will start chasing
+const stop_chase_distance: float = 150 #Distance to stop chasing
+
+func _physics_process(delta):
+    if !is_on_floor():
+        velocity.y += gravity * delta
+        velocity.x = 0
+    check_player_distance()
+    move(delta)
+    move_and_slide()
+
+#Determines if the enemy should be chasing, roaming, or dying
+func move(delta):
+    if !dead:
+        if !is_chase:
+            velocity += dir * speed * delta #wander if we arent chasing
```

Comment:
This acceleration here looks a bit strange with the velocity being set back to 0 every time the enemy timer while it's roaming reaches 0. I'd recommend either we go with 
`velocity = dir * speed * delta` where speed is just multiplied by like,, 80 or somethin
or we accelerate way faster, and then clamp the velocity so that it can't go higher than a certain amount.